app/controllers/queue_items_controller.rb

```diff
@@ -11,12 +11,43 @@ def create
   end
 
   def destroy
+    queue_item = QueueItem.find_by(id: params[:id], user: current_user)
+    if queue_item
+      queue_item.destroy
+      current_user.normalize_queue_item_positions
+    end
+    redirect_to my_queue_path
+  end
+
+  def update_queue
     begin
-      queue_item = QueueItem.find_by(id: params[:id], user: current_user)
-    rescue ActiveRecord::RecordNotFound
-    else
-      queue_item.destroy if queue_item
+      update_user_queue_attributes
+    rescue ActiveRecord::RecordInvalid
+      flash[:danger] = "Invalid position value."
+    else # ActiveRecord::RecordInvalid was not raised
+      current_user.normalize_queue_item_positions
     end
     redirect_to my_queue_path
   end
+
+  private
+
+  def update_user_queue_attributes
+    ActiveRecord::Base.transaction do
+      params[:queue_items].each do |queue_item_data|
+        queue_item = QueueItem.find queue_item_data[:id]
+        offset_position = get_new_offset_position queue_item_data[:position]
```

Comment:
Do not prefix `get_` for reader methods in ruby. Then every getter should start with `get_` and so on and the code would be bloated. Same goes for `set_`.